workspaces/core/src/combobox/Combobox.tsx

```diff
@@ -0,0 +1,170 @@
+import TextInput from '../textInput/TextInput';
+
+type Placement =
+  | 'top-start'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'right-start'
+  | 'right-end'
+  | 'left-start'
+  | 'left-end'
+  | 'top'
+  | 'bottom'
+  | 'right'
+  | 'left';
+
+type InputSize = 'sm' | 'md' | 'lg' | 'xl';
+type RenderProp = (data: { open?: boolean }) => ReactNode;
+type OptionsRenderProp = (data: {
+  active: boolean;
+  selected: boolean;
+  disabled: boolean;
+}) => ReactNode;
+
+type ComboboxRootProps = {
+  value: BaseOptionType;
+  onChange(value: T): void;
+  onQueryChange(value: string): void;
+  onClear?: () => void;
+  displayValue?: (value: T) => string;
+  isError?: boolean;
+  disabled?: boolean;
+  size?: InputSize;
+  className?: string;
+  position?: Placement;
+  children: ReactNode | RenderProp;
+  label?: string;
+  placeholder?: string;
+};
+
+type OptionProps = {
+  value: BaseOptionType;
+  displayRadioIcon?: boolean;
+  children?: ReactNode | OptionsRenderProp;
+};
+
+const ComboboxRoot: React.FC<ComboboxRootProps> = ({
+  children,
+  value,
+  onChange,
+  onQueryChange,
+  isError = false,
+  disabled = false,
+  size = 'md',
+  className,
+  position = 'bottom-start',
+  label,
+  placeholder,
+  displayValue,
+}) => {
+  const multiple = false;
+  const [anchorEl, setAnchorEl] = useState<Element | null>(null);
+  const [popperEl, setPopperEl] = useState<HTMLElement | null>(null);
+
+  const { styles, attributes } = usePopper(anchorEl, popperEl, {
+    placement: position,
+  });
+
+  const hasRenderProp = typeof children === 'function';
+
+  return (
+    <div className={mergeClassnames('w-full relative', className && className)}>
+      <HUICombobox
+        value={value}
+        onChange={onChange}
+        disabled={disabled}
+        multiple={multiple}
+        nullable
+      >
+        {({ open }) => (
+          <>
+            <div className="relative" ref={setAnchorEl}>
+              <HUICombobox.Input
+                onChange={({ target: { value } }) => onQueryChange(value)}
+                as={TextInput}
+                displayValue={displayValue}
+                label={label}
+                placeholder={placeholder}
+                inputSize={size}
+                type="text"
+                disabled={disabled}
+                isError={isError}
+              />
+              <HUICombobox.Button
+                className={mergeClassnames(
+                  'text-bulma absolute top-4 flex-shrink-0 transition-transform',
+                  size === 'sm' ? 'text-moon-16' : 'text-moon-24',
+                  size === 'xl' && 'top-4 right-4',
+                  size == 'md' && 'top-10 right-2',
+                  size === 'lg' && 'top-11 right-2',
+                  size === 'sm' && 'top-10 right-2',
+                  open && 'rotate-[-180deg]'
+                )}
+              >
+                <ControlsChevronDownSmall />
+              </HUICombobox.Button>
+            </div>
```

Comment:
this is better to move to subComponent Input, to have possibilities for customization with className